src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -32,68 +33,132 @@ static void Main(string[] args)
                    
                     if(_options.Output != null)
                     {
-                        if (!Directory.Exists(_options.Output))
+                        if (Path.GetExtension(_options.Output) != ".txt")
+                        {
+                            Console.Error.WriteLine("Error: output must be into a *.txt file");
+                            Environment.Exit(-3);
+                        }
+
+                        string directory = _options.Output.Substring(0, _options.Output.LastIndexOf("/", StringComparison.OrdinalIgnoreCase));
+
+                        if (!Directory.Exists(directory))
```

Comment:
If the user didn't specify any directory, just a filename, then I think directory will be null, so this should probably be:

`if ((directory != null) && !Directory.Exists(directory))`